functional/include/fun/option/option_inner.h

```diff
@@ -6,16 +6,16 @@
 namespace fun {
 
 //------------------------------------------------------------------------------
-template <class T, class En = void> struct OptionUnion;
+template <class T, class En = void> class OptionUnion;
 
 //------------------------------------------------------------------------------
 template <class T>
-struct OptionUnion<T, std::enable_if_t<std::is_empty<T>::value>>: private T {
+class OptionUnion<T, std::enable_if_t<std::is_empty<T>::value>>: T {
```

Comment:
Why switch these to classes?

Reply:
There started being some things I felt deserved to be private members, and even though `OptionUnion` is an "internal" type, since there are multiple specializations, making the public/private separation helped me to think about which parts of its API were expected to be defined.

Obviously wanting a public/private separation doesn't necessitate changing keywords, but when something starts having private members (especially private data members) I typically make it a `class`. It's my own personal style, I guess, that something is only a `struct` if its implementation and its API are one and the same.